sys/windows/syscall_windows.go

```diff
@@ -346,21 +346,17 @@ func GetDriveType(rootPathName string) (DriveType, error) {
 // https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getvolumeinformationw
 func GetFilesystemType(rootPathName string) (string, error) {
 	rootPathNamePtr, err := syscall.UTF16PtrFromString(rootPathName)
+	var systemType = "unavailable"
 	if err != nil {
 		return "", errors.Wrapf(err, "UTF16PtrFromString failed for rootPathName=%v", rootPathName)
 	}
-
 	buffer := make([]uint16, MAX_PATH+1)
+	// _GetVolumeInformation will fail for external drives like CD-ROM or other type with error codes as ERROR_NOT_READY. ERROR_INVALID_FUNCTION, ERROR_INVALID_PARAMETER, etc., these types of errors will be ignored
 	success, err := _GetVolumeInformation(rootPathNamePtr, nil, 0, nil, nil, nil, &buffer[0], MAX_PATH)
-	// check if CD-ROM or other type that is not supported in GetVolumeInformation function
-	if err == ERROR_NOT_READY || err == ERROR_INVALID_FUNCTION {
-		return "", nil
-	}
-	if !success {
-		return "", errors.Wrap(err, "GetVolumeInformationW failed")
+	if success {
```

Comment:
I feel like we should report `err` somewhere, if possible?

Reply:
we could do 2 things, either in the beats code check for `unavailable` and add an info message in the logs or add a note in the documentation regarding this specific field. I would go for the docs in this case, with this occasion maybe also clarify in some cases the field values in windows might have a different meaning . 
Either way I don't think we should complicate the code in the gosigar lib.

Reply:
Alright, that makes sense. Do you want to add the doc line in the PR to update `beats` ?

Reply:
yes, that will be a separate PR with the new version of gosigar

Reply:
Cool! I'm a little paranoid about weird errors/edge cases going undocumented, so I like to make sure.